test/HubPool.LiquidityProvision.ts

```diff
@@ -1,20 +1,124 @@
-    expect(await hubPool.callStatic.timerAddress()).to.equal(timer.address);
+  it("Adding ETH liquidity correctly wraps to WETH and mints LP tokens", async function () {
+    // Depositor can send WETH, if they have. Explicitly set the value to 0 to ensure we dont send any eth with the tx.
+    await weth.connect(liquidityProvider).approve(hubPool.address, amountToLp);
+    await hubPool.connect(liquidityProvider).addLiquidity(weth.address, amountToLp, { value: 0 });
+    expect(await weth.balanceOf(liquidityProvider.address)).to.equal(amountToSeedWallets.sub(amountToLp));
+    expect(await wethLpToken.balanceOf(liquidityProvider.address)).to.equal(amountToLp);
+
+    // Next, try depositing ETH with the transaction. No WETH should be sent. The ETH send with the TX should be
+    // wrapped for the user and LP tokens minted. Send the deposit and check the ether balance changes as expected.
+    await expect(() =>
+      hubPool.connect(liquidityProvider).addLiquidity(weth.address, amountToLp, { value: amountToLp })
+    ).to.changeEtherBalance(weth, amountToLp); // WETH's Ether balance should increase by the amount LPed.
+    // The weth Token balance should have stayed the same as no weth was spent.
+    expect(await weth.balanceOf(liquidityProvider.address)).to.equal(amountToSeedWallets.sub(amountToLp));
+    // However, the WETH LP token should have increase by the amount of LP tokens minted, as 2 x amountToLp.
+    expect(await wethLpToken.balanceOf(liquidityProvider.address)).to.equal(amountToLp.mul(2));
+    // Equally, the total WETH supply should have increased by the amount of LP tokens minted as they were deposited.
+    expect(await wethLpToken.totalSupply()).to.equal(amountToLp.mul(2));
+    expect(await weth.totalSupply()).to.equal(amountToLp.add(amountToSeedWallets));
+  });
+
+  it("Removing ETH liquidity can send back WETH or ETH depending on the users choice", async function () {
+    await weth.connect(liquidityProvider).approve(hubPool.address, amountToLp);
+    await hubPool.connect(liquidityProvider).addLiquidity(weth.address, amountToLp);
+
+    // Remove half the liquidity as WETH (set sendETH = false). This should modify the weth bal and not the eth bal.
+    await expect(() =>
+      hubPool.connect(liquidityProvider).removeLiquidity(weth.address, amountToLp.div(2), false)
+    ).to.changeEtherBalance(liquidityProvider, 0);
+
+    expect(await weth.balanceOf(liquidityProvider.address)).to.equal(amountToSeedWallets.sub(amountToLp.div(2))); // WETH balance should increase by the amount removed.
+
+    // Next, remove half the liquidity as ETH (set sendETH = true). This should modify the eth bal but not the weth bal.
+    await expect(() =>
+      hubPool.connect(liquidityProvider).removeLiquidity(weth.address, amountToLp.div(2), true)
+    ).to.changeEtherBalance(liquidityProvider, amountToLp.div(2)); // There should be ETH transferred, not WETH.
```

Comment:
this syntax is great! unfortunately it cant be chained along with a `.to.changeTokenBalance` so I did this the old fashion way.